src/methods.jl

```diff
@@ -77,3 +74,54 @@ julia> subtract_overscan(frame, (:, 4:5), dims = 2)
 * [`subtract_overscan!`](@ref)
 """
 subtract_overscan(frame::AbstractArray, idxs; dims = axes_min_length(idxs)) = subtract_overscan!(deepcopy(frame), idxs, dims = dims)
+
+
+"""
+    flat_correct!(frame::AbstractArray, flat_frame::AbstractArray; norm_value = mean(flat_frame))
+
+In place version of [`flat_correct`](@ref)
+"""
+function flat_correct!(frame::AbstractArray, flat_frame::AbstractArray; norm_value = mean(flat_frame))
+    size(frame) != size(flat_frame) && error("size of frame and flat frame are not same")
+    norm_value < 0 && error("norm_value must be greater than 0")
+
+    frame ./= (flat_frame ./ norm_value)
+    return frame
+end
+
+
+"""
+    flat_correct(frame::AbstractArray, flat_frame::AbstractArray; norm_value = mean(flat_frame))
+
+Correct `frame` for non-uniformity using the calibrated `flat_frame`.
+
+By default, the `flat_frame` is normalized by its mean, but this can be changed by providing a custom `norm_value`.
+
+!!! note
+    This function may introduce non-finite values if `flat_frame` contains values very close to `0` due to dividing by zero.
+    The default behavior will return `Inf` if the frame value is non-zero, and `Nan` if the frame value is `0`.
+
+# Examples
+```jldoctest
+julia> frame = ones(3, 3);
+
+julia> flat = fill(2.0, (3, 3));
+
+julia> flat_correct(frame, flat, norm_value = 1.0)
+3×3 Array{Float64,2}:
+ 0.5  0.5  0.5
+ 0.5  0.5  0.5
+ 0.5  0.5  0.5
+
+julia> flat_correct(frame, flat)
+3×3 Array{Float64,2}:
+ 1.0  1.0  1.0
+ 1.0  1.0  1.0
+ 1.0  1.0  1.0
+
+```
+
+# See Also
+* [`flat_correct!`](@ref)
+"""
+flat_correct(frame::AbstractArray, flat_frame::AbstractArray; norm_value = mean(flat_frame)) = flat_correct!(deepcopy(frame), flat_frame, norm_value = norm_value)
```

Comment:
Here I'd do:
```suggestion
flat_correct(frame::AbstractArray, flat_frame::AbstractArray; kwargs...) = flat_correct!(deepcopy(frame); kwargs...)
```
The advantage is that the list of keyword arguments is automatically inherited from `flat_correct!` and you don't need to manually keep them consistent